### ISSUE TITLE
tweak check a bit when we enable auto-pagingt: turn it off if when stderr doesn't match stdin's tty too

### ISSUE THREAD

**Issue by Skarlso** [systemd/systemd#9192]
systemd version
```bash
systemd --v
systemd 229
+PAM +AUDIT +SELINUX +IMA +APPARMOR +SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ -LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD -IDN
```

If I call loginctl with an unknown / not logged in user to show the linger flag it throws an error like this:

```bash
DEBUG: Failed to get user: No user 10098 known or logged in
```

Now, notice how I can't suppress this message only by directing both & to /dev/null:

```bash
loginctl show-user 't0003s121' --property=Linger
Failed to get user: No user 10098 known or logged in

loginctl show-user 't0003s121' --property=Linger 1>/dev/null
Failed to get user: No user 10098 known or logged in

loginctl show-user 't0003s121' --property=Linger 2>/dev/null
Failed to get user: No user 10098 known or logged in

loginctl show-user 't0003s121' --property=Linger &>/dev/null
```

Obviously it's not good to suppress out since on a further check I'm relying on the output saying `Linger=yes` or `Linger=no` in case the user happens to be logged in at the moment.

This is incredibly annoying because I would like to enable linger in an automated way but we have checks which look for errors in syslog which fail because of this error. And I can't seem to make it so that it doesn't display this error. I'm relying in the return value of this command and the output with a grep. Which is, when it's 1, I'm enabling linger for that user. If it's 0 all is well.

The full check is for example:

```bash
loginctl show-user '${user}' --property=Linger | grep -q -e 'Linger=yes'
```

Is there a flag I could use so it remains quiet?

I'm aware of the implementation detail that users are under /something/something/linger. But I would very much like to not depend on an implementation detail that could possibly change since it's not a technical requirement.

**Comment by poettering** [comment]
the error message is actually written to stderr, while the property itself is written to stdout. However, loginctl (as many of systemd's tools, as well as other more modern stuff, such as git) default to auto-paging, i.e. longer output (both stdout + stderr) is automatically fed to $PAGER or "less". That happens only if stdout is a tty though. By redirecting stdout but not stderr you hence turn off the pager, but redirecting only stderr, but not stdout the pager is on, but since we redirect stderr to the pager this is without effect....

I figure we should tweak the logic when we turn on auto-paging and also turn it off when stderr is not a tty, or at least leave it in place if it doesn't match the same tty as stdout...

Anyway, a quick fix for your problem is to add --no-pager to your command or set $PAGER=cat, both of which will force the pager to be turned off...

**Comment by Skarlso** [comment]
Oh wow. :O I actually looked at that setting and was wondering if it's related, but failed to make this connection.

Thanks! That worked!

**Comment by poettering** [comment]
Fix pending in #9202
